src/udx.c

```diff
@@ -644,8 +644,10 @@ process_packet (udx_socket_t *socket, char *buf, ssize_t buf_len, struct sockadd
 
   // We expect this to be a stream packet from now on
 
-  if (!(stream->status & UDX_STREAM_CONNECTED) && (stream->on_firewall != NULL && stream->on_firewall(stream, socket, addr))) {
-    return 1;
+  if (!(stream->status & UDX_STREAM_CONNECTED)) {
+    if (stream->on_firewall != NULL && stream->on_firewall(stream, socket, addr)) return 1;
```

Comment:
```suggestion
    if (stream->on_firewall == NULL || stream->on_firewall(stream, socket, addr)) return 1;
```

Early bail if the firewall hook isn't set.

Reply:
right!